server.js

```diff
@@ -1,30 +1,129 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+import express, { response } from "express";
+import cors from "cors";
+import mongoose from "mongoose";
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+const port = process.env.PORT || 8080;
+const paginate = require('express-paginate')
+const app = express();
+
+// to create users 
+const User= mongoose.model('Users',{
+  username:{
+    type: String,
+    lowercase: true,
+    unique: true
+  }
+})
+
+//to create Thoughts 
+const Thought = mongoose.model("Thought", {
+  message: {
+    type: String,
+    required: true,
+    minLength: 5,
+    maxLength:140
+  },
+  hearts: {
+    type: Number,
+    default: 0,
+  },
+  createDate: {
+    type: Date,
+    default: Date.now,
+  },
+  username:{
+    type: String,
+    ref: User,
+    required: true
+  },
+  hashtag:{
+    type: Array,
+    default: []
+  }
+});
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(express.json());
+app.use(paginate.middleware(20, 50))
+
+// Get  all Thoughts
+app.get("/", async (req, res) => {
+  try{
+    const allThoughts = await Thought.find().limit(req.query.limit).sort({createDate: -1})
+    res.json(allThoughts)
+  }catch(error){
+    res.status(400).json(error)
+  }
+});
+
+//Create user
+app.post("/user/create", async (req, res)=>{
+  try {
+    const newUser = await new User({username: req.body.username}).save()
+    res.json(newUser)  
+  } catch (error) {
+    res.status(400).json(error)
+  }
+})
+//search users
+app.get("/users", async (req, res)=>{
+  try {
+    const allUser= await User.find()
+    res.json(allUser)
+  } catch (error) {
+    res.status(400).json(error)
+  }
+})
```

Comment:
Nice that you provided a simple user system, now after auth project you can secure them 🤓